proof/verifier.go

```diff
@@ -180,71 +226,276 @@ func (p *Proof) verifySplitRootProof() error {
 	return err
 }
 
-// verifyExclusionProofs verifies all ExclusionProofs are valid.
 func (p *Proof) verifyExclusionProofs() (*commitment.TapCommitmentVersion,
 	error) {
 
-	// Gather all P2TR outputs in the on-chain transaction.
-	p2trOutputs := make(map[uint32]struct{})
+	// There is nothing to check so return early.
+	if p.ExclusionProofs == nil {
+		return nil, nil
+	}
+
+	// Gather all P2TR outputs in the on-chain transaction. The STXO proofs
+	// are tracked using a set of serialized keys. We ignore that set when
+	// verifying V1 exclusion proofs.
+	p2trOutputs := make(map[uint32]fn.Set[asset.SerializedKey])
 	for i, txOut := range p.AnchorTx.TxOut {
 		if uint32(i) == p.InclusionProof.OutputIndex {
 			continue
 		}
-		if txscript.IsPayToTaproot(txOut.PkScript) {
-			p2trOutputs[uint32(i)] = struct{}{}
+		if !txscript.IsPayToTaproot(txOut.PkScript) {
+			continue
 		}
+
+		outIdx := uint32(i)
+		p2trOutputs[outIdx] = make(fn.Set[asset.SerializedKey])
```

Comment:
Hmm, we don't actually add any keys to this set here?

Reply:
With the refactor of `p2trOutputs` we now use a simple `fn.Set[uint32]` here and only set the keys once we are verifying and tracking stxo proofs.